src/main/java/de/dhbw/mh/rinne/ast/AstFunctionCallNode.java

```diff
@@ -1,16 +1,23 @@
 package de.dhbw.mh.rinne.ast;
 
+import java.util.List;
+
 import de.dhbw.mh.rinne.CodeLocation;
 
 public class AstFunctionCallNode extends AstExpressionNode {
+    private final String functionName;
+    private final List<AstExpressionNode> arguments;
 
-    public AstFunctionCallNode(CodeLocation codeLocation) {
+    public AstFunctionCallNode(CodeLocation codeLocation, String functionName, List<AstExpressionNode> arguments) {
         super(codeLocation);
+        this.functionName = functionName;
+        this.arguments = arguments;
+        arguments.forEach(e -> this.children.add(e));
```

Comment:
Unless you have a specific reason for populating `children` manually, direct assignment keeps the code simpler and avoids duplication.

```suggestion
        this.arguments = arguments;
```